### PR TITLE
feat: [macOS] Add support for TextBox

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

Adding a `TextBox` is crashing the app.

## What is the new behavior?

- The `TextBox` now shows up and updates its `Text` property. There's a lot left to do, but this unblocks the ability to display a `TextBox`.
- Fix nuget dev override on macOS
- `TextElement.Background` cannot be set to null
- Adjust native styles defaults
- Fixed `DefaultPreferredBodyFontSize` to `SystemFontSize`, thanks @MartinZikmund 
- Use `NSView.FittingSize` as fallback measure

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Related to #626

### CODE REVIEW

**Comment by MartinZikmund** [comment]
I have a [partial implementation](https://github.com/unoplatform/uno/compare/master...MartinZikmund:dev/mazi/textbox-macos#diff-f210465cc1ae690bfe19c5abbceffcc5L23) as well, so maybe some things can be reused here:

- `NSFontHelper` - change of default font size
- [This commit](https://github.com/unoplatform/uno/commit/8fbd9e6cbf77808a065e3f29f66214ae6a59bde9) - it sets macOS specific native classes as views where it is not yet done
-  Set `DrawsBackground = false;` in `ScrollContentPresenter` to avoid dark background set by system when OS is in dark mode
- In [`SingleLineTextboxView`](https://github.com/unoplatform/uno/compare/master...MartinZikmund:dev/mazi/textbox-macos#diff-3b2cffe6038e76656e9f70df593c79b4) is an implementation for setting insertion caret foreground color

**Comment by jeromelaban** [comment]
Thanks for the updates! I'll merge your changes in this PR.

**Comment by MartinZikmund** [comment]
@jeromelaban not all of them though, my branch is completely separate implementation, so I maybe cherry pick things that are useful only 😄

File: src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs [hunk 1]
```diff
@@ -1,17 +1,164 @@
...
 {
-	public partial class ScrollViewer 
+	public partial class ScrollViewer : ContentControl
 	{
-		private void UpdateZoomedContentAlignment() { }
+		/// <summary>
+		/// On iOS 10-, we set a flag on the view controller such that the CommandBar doesn't automatically affect ScrollViewer content 
+		/// placement. On iOS 11+, we set this behavior on the UIScrollView itself.
+		/// </summary>
+		internal static bool UseContentInsetAdjustmentBehavior => false;
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs:25]: The comment doesn't apply, and probably the property doesn't apply either?

File: src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs [hunk 2]
```diff
@@ -1,17 +1,164 @@
...
+using System.Linq;
+
+using Foundation;
+using CoreGraphics;
+using Uno.UI.Extensions;
 
 namespace Windows.UI.Xaml.Controls
 {
-	public partial class ScrollViewer 
+	public partial class ScrollViewer : ContentControl
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs:19]: What's needed from this code?

File: src/SourceGenerators/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.csproj [hunk 3]
```diff
@@ -94,7 +94,10 @@
 
 	<Target Name="_UnoToolkitOverrideNuget" AfterTargets="AfterBuild" DependsOnTargets="BuiltProjectOutputGroup" Condition="'$(UnoNugetOverrideVersion)'!=''">
 		<PropertyGroup>
-			<_TargetNugetFolder>$(USERPROFILE)\.nuget\packages\Uno.UI\$(UnoNugetOverrideVersion)\tools</_TargetNugetFolder>
+			<_baseNugetPath Condition="'$(USERPROFILE)'!=''">$(USERPROFILE)</_baseNugetPath>
+      <_baseNugetPath Condition="'$(HOME)'!=''">$(HOME)</_baseNugetPath>
+
+			<_TargetNugetFolder>$(_baseNugetPath)\.nuget\packages\Uno.UI\$(UnoNugetOverrideVersion)\tools</_TargetNugetFolder>
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.csproj:100]: it does!

[the remark this answers is not in the document]
Comment (davidjohnoliver) [reply src/SourceGenerators/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.csproj:100]: woohoo!

File: src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs:25]: Indeed, removed.

File: src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.macOS.cs:19]: Not useful, removed.

**Pull request merged**